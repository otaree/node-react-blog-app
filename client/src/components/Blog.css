.Blog {
    width: 80%;
    margin: 10px auto;
    padding: 10px 20px;
    box-sizing: border-box;
}

.Blog h2 {
    text-align: center;
}

.btn-edit {
    outline: none;
    color: white;
    border: 1px solid #eee;
    border-radius: 3px;
    width: 70px;
    height: 30px;
    font-size: 16px;
    font-weight: bold;
    background-color: rgb(128, 187, 40);
}

.btn-edit:hover {
    color: #444;
    background: greenyellow;
}

.btn-edit a {
    text-decoration: none;
    color: inherit;
}

