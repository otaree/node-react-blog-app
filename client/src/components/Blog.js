import React from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import "./Blog.css";

export const Blog = props => {
    const id = props.match.params.id;
    const blog = props.blogs.find(blog => blog._id === id);

    if (!blog) {
        return (
            <section>
                <div>Sorry, but no blog was found.</div>
            </section>
        );
    }

    return (
        <section className="Blog">
            <h2>{blog.title}</h2>
            <p>{blog.body}</p>
            <p>{blog.author}</p>
            <div className="Blog __body">
                {blog.published && <p>Published: {blog.publishedAt}</p>}
            </div>
            <div>
                <button className="btn-edit">
                    <Link to={`/blogs/edit/${id}`}>
                        Edit
                    </Link>
                </button>
            </div>
        </section>
    );
}

const mapStateToProps = state => {
    return {
        blogs: state
    };
};

export default connect(mapStateToProps)(Blog);