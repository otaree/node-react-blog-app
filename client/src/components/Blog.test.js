import React from 'react';
import { shallow } from 'enzyme';
import { Blog } from './Blog';

describe('Blog', () => {
    let props = {
        blogs: [
            {
                _id: '1',
                title: "First Title",
                author: "Tester",
                body: "First's body",
                published: false
            },
            {
                _id: '2',
                title: "Second Title",
                author: "Tester",
                body: "Second's body",
                published: true,
                publishedAt: 1000
            }
        ],
        match: {
            params: {
                id: '2'
            }
        }
    };

    let blog = shallow(<Blog {...props} />);

    it('renders properly', () => {
        expect(blog).toMatchSnapshot();
    });

    it('displays the correct blog title', () => {
        expect(blog.find('h2').text()).toEqual(props.blogs[1].title);
    });

    it('displays edit button', () => {
        expect(blog.find('.btn-edit').length).toEqual(1);
    });
});