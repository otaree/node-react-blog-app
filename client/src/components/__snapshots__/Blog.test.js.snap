// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Blog renders properly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    blogs={
        Array [
            Object {
              "_id": "1",
              "author": "Tester",
              "body": "First's body",
              "published": false,
              "title": "First Title",
            },
            Object {
              "_id": "2",
              "author": "Tester",
              "body": "Second's body",
              "published": true,
              "publishedAt": 1000,
              "title": "Second Title",
            },
          ]
    }
    match={
        Object {
            "params": Object {
              "id": "2",
            },
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2>
          Second Title
</h2>,
        <p>
          Second's body
</p>,
        <p>
          Tester
</p>,
        <div
          className="Blog __body"
>
          <p>
                    Published: 
                    1000
          </p>
</div>,
        <div>
          <button
                    className="btn-edit"
          >
                    <Link
                              replace={false}
                              to="/blogs/edit/2"
                    >
                              Edit
                    </Link>
          </button>
</div>,
      ],
      "className": "Blog",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Second Title",
        },
        "ref": null,
        "rendered": "Second Title",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Second's body",
        },
        "ref": null,
        "rendered": "Second's body",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Tester",
        },
        "ref": null,
        "rendered": "Tester",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <p>
            Published: 
            1000
</p>,
          "className": "Blog __body",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Published: ",
              1000,
            ],
          },
          "ref": null,
          "rendered": Array [
            "Published: ",
            1000,
          ],
          "type": "p",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <button
            className="btn-edit"
>
            <Link
                        replace={false}
                        to="/blogs/edit/2"
            >
                        Edit
            </Link>
</button>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Link
              replace={false}
              to="/blogs/edit/2"
>
              Edit
</Link>,
            "className": "btn-edit",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Edit",
              "replace": false,
              "to": "/blogs/edit/2",
            },
            "ref": null,
            "rendered": "Edit",
            "type": [Function],
          },
          "type": "button",
        },
        "type": "div",
      },
    ],
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2>
            Second Title
</h2>,
          <p>
            Second's body
</p>,
          <p>
            Tester
</p>,
          <div
            className="Blog __body"
>
            <p>
                        Published: 
                        1000
            </p>
</div>,
          <div>
            <button
                        className="btn-edit"
            >
                        <Link
                                    replace={false}
                                    to="/blogs/edit/2"
                        >
                                    Edit
                        </Link>
            </button>
</div>,
        ],
        "className": "Blog",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Second Title",
          },
          "ref": null,
          "rendered": "Second Title",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Second's body",
          },
          "ref": null,
          "rendered": "Second's body",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Tester",
          },
          "ref": null,
          "rendered": "Tester",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <p>
              Published: 
              1000
</p>,
            "className": "Blog __body",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Published: ",
                1000,
              ],
            },
            "ref": null,
            "rendered": Array [
              "Published: ",
              1000,
            ],
            "type": "p",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <button
              className="btn-edit"
>
              <Link
                            replace={false}
                            to="/blogs/edit/2"
              >
                            Edit
              </Link>
</button>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <Link
                replace={false}
                to="/blogs/edit/2"
>
                Edit
</Link>,
              "className": "btn-edit",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Edit",
                "replace": false,
                "to": "/blogs/edit/2",
              },
              "ref": null,
              "rendered": "Edit",
              "type": [Function],
            },
            "type": "button",
          },
          "type": "div",
        },
      ],
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
